test/components/componentTests.ts

```diff
@@ -246,333 +344,338 @@ describe("Component behavior", () => {
+      let c = new Plottable.Component();
+      c.renderImmediately = () => {
+        renderFlag = true;
+        return c;
+      };
+      c.anchor(svg);
+      c.render();
+      assert.isFalse(renderFlag, "no render until width/height set to nonzero");
+
+      let offeredWidth = 10;
+      let offeredHeight = 0;
+      c.computeLayout({x: 0, y: 0}, offeredWidth, offeredHeight);
+      c.render();
+      assert.isTrue(renderFlag, "render still occurs if one of width/height is zero");
+
+      renderFlag = false;
+      offeredHeight = 10;
+      c.computeLayout({x: 0, y: 0}, offeredWidth, offeredHeight);
+      c.render();
+      assert.isTrue(renderFlag, "render occurs if width and height are positive");
+
+      c.destroy();
+      svg.remove();
+    });
   });
 
-  it("renderTo() only accepts strings, selections containing svgs, and SVG elements", () => {
-    svg.attr("id", "render-to-test");
-    assert.doesNotThrow(() => c.renderTo("#render-to-test"), Error, "accepts strings that identify svgs");
-    assert.doesNotThrow(() => c.renderTo(svg), Error, "accepts selections that contain svgs");
-    assert.doesNotThrow(() => c.renderTo(document.getElementById("render-to-test")), Error, "accepts svg elements");
-    let parent = TestMethods.getSVGParent();
-    let div = parent.append("div");
-    // HACKHACK #2614: chai-assert.d.ts has the wrong signature
-    (<any> assert).throws(() => c.renderTo(div), Error,
-      "Plottable requires a valid SVG to renderTo", "rejects selections that don't contain svgs");
-    (<any> assert).throws(() => c.renderTo(<Element> div.node()), Error,
-      "Plottable requires a valid SVG to renderTo", "rejects DOM nodes that are not svgs");
-    (<any> assert).throws(() => c.renderTo("#not-a-element"), Error,
-      "Plottable requires a valid SVG to renderTo", "rejects strings that don't correspond to DOM elements");
-    (<any> assert).throws(() => c.renderTo(d3.select(null)), Error,
-      "Plottable requires a valid SVG to renderTo", "rejects empty d3 selections");
-    svg.remove();
-  });
+  describe("rendering to a DOM node", () => {
 
-  describe("origin methods", () => {
-    let cWidth = 100;
-    let cHeight = 100;
+    let renderFlag: boolean;
 
-    it("modifying returned value does not affect origin", () => {
-      c.renderTo(svg);
-      let receivedOrigin = c.origin();
-      let delta = 10;
-      receivedOrigin.x += delta;
-      receivedOrigin.y += delta;
-      assert.notStrictEqual(receivedOrigin.x, c.origin().x, "receieved point can be modified without affecting origin (x)");
-      assert.notStrictEqual(receivedOrigin.y, c.origin().y, "receieved point can be modified without affecting origin (y)");
+    beforeEach(() => {
+      renderFlag = false;
+      c.renderImmediately = () => {
+        renderFlag = true;
+        return this;
+      };
+    });
+
+    it("rendering to a DOM node involves anchoring, layout computing, and actual rendering", () => {
```

Comment:
anchor(), computeLayout(), render()
also, can you change the test case description so that it reads semantically?